From bc1af8fc831eeaf32c51197e3603c89b08caf97e Mon Sep 17 00:00:00 2001
From: jchialun <johnny.chialung@windriver.com>
Date: Fri, 10 Oct 2025 12:18:54 -0500
Subject: [PATCH 1/1] Volume backup support with Openstack plugging and SDK

Add openstack plugin support to communicate with STX-O and Openstack
vanilla by adding a default value to the location parameter. It is
required to keep compatibility with vanilla versions.

Signed-off-by: Johnny Chia <johnny.chialung@windriver.com>
---
 openstackclient/volume/v2/volume_backup.py | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/openstackclient/volume/v2/volume_backup.py b/openstackclient/volume/v2/volume_backup.py
index decb5beb..bf8646ea 100644
--- a/openstackclient/volume/v2/volume_backup.py
+++ b/openstackclient/volume/v2/volume_backup.py
@@ -158,9 +158,13 @@ class CreateVolumeBackup(command.ShowOne):
 
             kwargs['metadata'] = parsed_args.properties
 
-        if not parsed_args.location and parsed_args.properties:
-            parsed_args.location = parsed_args.properties.get("location", None)
-        elif sdk_utils.supports_microversion(volume_client, '3.43'):
+        if not parsed_args.location:
+            if parsed_args.properties:
+                parsed_args.location = parsed_args.properties.get("location", "ceph")
+            else:
+                parsed_args.location = "ceph"
+
+        if sdk_utils.supports_microversion(volume_client, '3.43'):
             if kwargs.get('metadata', None):
                 kwargs['metadata']['location'] = parsed_args.location
             else:
-- 
2.43.0

